Use iris flower dataset from sklearn library and use cross_val_score against following models to measure the performance of each. In the end figure out the model with best performance,

- Logistic Regression
- SVM
- Decision Tree
- Random Forest

In [1]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
import numpy as np
from sklearn.datasets import load_iris

In [2]:
iris = load_iris()

In [3]:
from sklearn.model_selection import cross_val_score

Logistic Regression

In [4]:
l_score = cross_val_score(LogisticRegression(solver='liblinear',multi_class='ovr'), iris.data, iris.target, cv=3)
np.average(l_score)

0.9533333333333333

Decision Tree

In [5]:
d_score =cross_val_score(DecisionTreeClassifier(), iris.data, iris.target, cv=3)
np.average(d_score)

0.9733333333333333

Support Vector Machine (SVM)

In [6]:
s_score = cross_val_score(SVC(), iris.data, iris.target, cv=3)
np.average(s_score)

0.96

Random Forest

In [7]:
r_score = cross_val_score(RandomForestClassifier(), iris.data, iris.target, cv=3)
np.average(r_score)

0.96

Best score so far is from Decision Tree: 0.9733333333333333